app/quest-boost/[boostId]/page.tsx

```diff
@@ -0,0 +1,193 @@
+import { Call, CallDetails } from "starknet";
+import {
+  getBoostById,
+  getQuestBoostClaimParams,
+  getQuestParticipants,
+  getQuestsInBoost,
+  updateQuestBoostClaimStatus,
+} from "../../../services/apiService";
+import Quest from "../../../components/quests/quest";
+import { useRouter } from "next/navigation";
+import { QuestDocument } from "../../../types/backTypes";
+import Timer from "../../../components/quests/timer";
+import { useAccount, useContractWrite } from "@starknet-react/core";
+import Button from "../../../components/UI/button";
+import boostContractCalls from "../../../utils/calldata/boost_contract_claim";
+import { useNotificationManager } from "../../../hooks/useNotificationManager";
+import {
+  NotificationType,
+  TransactionType,
+} from "../../../constants/notifications";
+import { hexToDecimal } from "../../../utils/feltService";
+import { CDNImage } from "../../../components/cdn/image";
+
+type BoostQuestPageProps = {
+  params: {
+    boostId: string;
+  };
+};
+
+export default function Page({ params }: BoostQuestPageProps) {
+  const router = useRouter();
+  const { address } = useAccount();
+  const { boostId } = params;
+  const [quests, setQuests] = useState([] as QuestDocument[]);
+  const [boost, setBoost] = useState({} as Boost);
+  const [participants, setParticipants] = useState<number>();
+  const [calls, setCalls] = useState<CallDetails>({} as CallDetails);
+  const [sign, setSign] = useState<string[]>([]); // [r, s]
+  const { addTransaction } = useNotificationManager();
+
+  const getTotalParticipants = async (questIds: number[]) => {
+    let total = 0;
+    await Promise.all(
+      questIds?.map(async (questID) => {
+        const res = await getQuestParticipants(questID);
+        if (res?.count) total += res?.count;
+      })
+    );
+    return total;
+  };
+
+  const fetchPageData = async () => {
+    const questsList = await getQuestsInBoost(boostId);
+    const boostInfo = await getBoostById(boostId);
+    const totalParticipants = await getTotalParticipants(boostInfo.quests);
+    setQuests(questsList);
+    setBoost(boostInfo);
+    setParticipants(totalParticipants);
+  };
+
+  useEffect(() => {
+    fetchPageData();
+  }, []);
+
+  const fetchBoostClaimParams = async () => {
+    try {
+      if (!boost?.id || !address) return;
+      const res = await getQuestBoostClaimParams(boost.id);
+      const formattedSign = [res?.r, res?.s];
+      return formattedSign;
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  useEffect(() => {
+    if (!boost?.id || !(sign.length > 0)) return;
+    const data = boostContractCalls.boostContractClaimData(
+      hexToDecimal(process.env.NEXT_PUBLIC_QUEST_BOOST_CONTRACT),
+      boost.id,
+      boost.amount,
+      boost.token,
+      sign
+    );
+    setCalls(data);
+  }, [boost, sign]);
+
+  //Contract
+  const {
+    data,
+    error,
+    writeAsync: execute,
+  } = useContractWrite({ calls: [calls as Call] });
```

Comment:
so here you'll be able to remove the `as Call`

Reply:
this gives a type error on sending with `CallDetails` type it fails. It expects `Call` type so i need to force the type here so that the linter doesn't highlight this